test/state/system_contracts.hpp

```diff
@@ -18,15 +19,29 @@ constexpr auto BEACON_ROOTS_ADDRESS = 0x000F3df6D732807Ef1319fB7B8bB8522d0Beac02
 /// The address of the system contract storing historical block hashes (EIP-2935).
 constexpr auto HISTORY_STORAGE_ADDRESS = 0x0aae40965e6800cd9b1f4b05ff21581047e3f91e_address;
 
+/// The address of the system contract processing EL-triggerable withdrawals (EIP-7002).
+constexpr auto WITHDRAWAL_REQUESTS_ADDRESS = 0x09Fc772D0857550724b07B850a4323f39112aAaA_address;
+
+/// The address of the system contract processing consolidations (EIP-7251).
+constexpr auto CONSOLIDATION_REQUEST_ADDRESS = 0x01aBEa29659e5e97C95107F20bb753cD3e09bBBb_address;
+
 struct BlockInfo;
 struct StateDiff;
 class BlockHashes;
 class StateView;
 
-/// Performs the system call: invokes system contracts.
+/// Performs the system call: invokes system contracts at the start of the block.
 ///
 /// Executes code of pre-defined accounts via pseudo-transaction from the system sender (0xff...fe).
 /// The sender's nonce is not increased.
-[[nodiscard]] StateDiff system_call(const StateView& state_view, const BlockInfo& block,
+[[nodiscard]] StateDiff system_call_block_start(const StateView& state_view, const BlockInfo& block,
     const BlockHashes& block_hashes, evmc_revision rev, evmc::VM& vm);
+
+/// Performs the system call: invokes system contracts at the end of the block.
+///
+/// Executes code of pre-defined accounts via pseudo-transaction from the system sender (0xff...fe).
+/// The sender's nonce is not increased.
```

Comment:
add a 1-sentence summary explanation why these are separate. Judging by just the API, they wouldn't need to be separate, because Requests vector could be returned empty, so I think leaving a hint why is worthwhile.

Reply:
I didn't know why not have one function with just empty requests returned at block start without looking inside the implementation and comparing. Actually, I'm not 100% sure even now :thinking: why?

Reply:
It's different contracts called at the start and end of block...

Reply:
> It's different contracts called at the start and end of block...

oh, lol, I hallucinated that the contracts list is an input to the function, but yea it's hardcoded... anyhow, a rephrasing could help I guess, maybe sth like `invokes _the_ system contracts _configured to run_ at the start of the block`. Same as the refactor you've mentioned elsewhere, let's see...

Reply:
The changes addressed my concerns, thank you!